viam_cartographer.go

```diff
@@ -225,14 +240,18 @@ func New(
 		name:         lidarName,
 		dataRateMsec: optionalConfigParams.LidarDataRateMsec,
 		actual:       lidarObject,
-		testing:      timedLidar,
+		timed:        timedLidar,
 	}
 
 	imu := IMU{
 		name:         optionalConfigParams.ImuName,
 		dataRateMsec: optionalConfigParams.ImuDataRateMsec,
 		actual:       imuObject,
-		testing:      timedIMU,
+		timed:        timedIMU,
+	}
+
+	if lidar.dataRateMsec == 0 && imu.dataRateMsec != 0 {
+		logger.Warn("In offline mode, but IMU data frequency is nonzero")
```

Comment:
Should this be an error?

Reply:
I can see it both ways, either it should error out as it is an incongruity in the config, or we want to warn the user that we will override the imu data rate as the lidar is the primary sensor. What are your thoughts?

Reply:
I think this case should error out and only override in the case that  `lidar.dataRateMsec == 0` and `imu.dataRateMsec` is not provided